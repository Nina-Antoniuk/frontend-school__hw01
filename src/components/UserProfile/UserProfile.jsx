import React from 'react';
import { PropTypes } from 'prop-types';
import s from './UserProfile.module.scss';

const UserProfile = function UserProfile({ profileInfo: { user, stats } }) {
  return (
    <div className={s.userProfile}>
      <div className={s.thumb}>
        <img
          className={s.userAvatar}
          src={
            (user && user.avatarMedium) || 'https://via.placeholder.com/300x300'
          }
          alt={(user && user.nickname) || 'placeholder for staranger'}
          width="300"
          height="300"
        />
      </div>
      <div className={s.userInfo}>
        <p>
          <span className={s.field}>User name</span>:
          {(user && user.nickname) || 'staranger'}
        </p>
        <p>
          <span className={s.field}>Followers</span>:
          {(stats && stats.followerCount) || 0}
        </p>
        <p>
          <span className={s.field}>Following</span>:
          {(stats && stats.followingCount) || 0}
        </p>
        <p>
          <span className={s.field}>Count of posts</span>:
          {(stats && stats.videoCount) || 0}
        </p>
        <p>
          <span className={s.field}>Signature</span>:
          {(user && user.signature) || ''}
        </p>
      </div>
    </div>
  );
};

UserProfile.defaultProps = {
  profileInfo: {
    user: {
      nickname: '',
      avatarMedium: '',
      signature: '',
    },
    stats: {},
  },
};

UserProfile.propTypes = {
  profileInfo: PropTypes.shape({
    user: PropTypes.shape({
      avatarMedium: PropTypes.string,
      nickname: PropTypes.string,
      signature: PropTypes.string,
    }),
    stats: PropTypes.shape({
      followerCount: PropTypes.number,
      followingCount: PropTypes.number,
      videoCount: PropTypes.string,
    }),
  }),
};

export default UserProfile;
