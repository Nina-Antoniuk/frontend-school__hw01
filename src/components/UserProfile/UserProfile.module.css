.userProfile {
  padding: 20px;
  -webkit-box-shadow: var(--shadow) 0px 5px 15px;
  box-shadow: var(--shadow) 0px 5px 15px;
  background-color: var(--light-color);
}

@media screen and (max-width: 767.98px) {
  .userProfile {
    max-width: 300px;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 30px;
  }
}

@media screen and (min-width: 768px) {
  .userProfile {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    max-width: 500px;
    margin-left: 30%;
    margin-bottom: 50px;
    padding: 80px 50px 30px;
  }
}

@media screen and (min-width: 1200px) {
  .userProfile {
    max-width: 800px;
  }
}

.thumb {
  max-width: 250px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  background-color: var(--placeholder-color);
}

@media screen and (min-width: 768px) {
  .thumb {
    max-width: 300px;
  }
}

.userAvatar {
  display: block;
  width: 100%;
}

@media screen and (min-width: 1200px) {
  .userInfo {
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
  }
}

@media screen and (min-width: 768px) {
  .description {
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    margin-top: 30px;
  }
}

.field {
  text-decoration: underline;
}
/*# sourceMappingURL=UserProfile.module.css.map */
