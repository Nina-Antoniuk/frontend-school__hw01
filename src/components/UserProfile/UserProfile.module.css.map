{
    "version": 3,
    "mappings": "AAEA,AAAA,YAAY,CAAC;EACX,OAAO,EAAE,IAAI;EACb,UAAU,EAAE,aAAa,CAAC,GAAG,CAAC,GAAG,CAAC,IAAI;EACtC,gBAAgB,EAAE,kBAAkB;CAsBrC;;AApBC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,QAAQ;EALxC,AAAA,YAAY,CAAC;IAMT,SAAS,EAAE,KAAK;IAChB,WAAW,EAAE,IAAI;IACjB,YAAY,EAAE,IAAI;IAClB,aAAa,EAAE,IAAI;GAgBtB;;;AAbC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EAZrC,AAAA,YAAY,CAAC;IAaT,OAAO,EAAE,IAAI;IACb,SAAS,EAAE,IAAI;IACf,WAAW,EAAE,MAAM;IACnB,SAAS,EAAE,KAAK;IAChB,WAAW,EAAE,GAAG;IAChB,aAAa,EAAE,IAAI;IACnB,OAAO,EAAE,cAAc;GAM1B;;;AAHC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,MAAM;EAtBtC,AAAA,YAAY,CAAC;IAuBT,SAAS,EAAE,KAAK;GAEnB;;;AAED,AAAA,MAAM,CAAC;EACL,SAAS,EAAE,KAAK;EAChB,WAAW,EAAE,IAAI;EACjB,YAAY,EAAE,IAAI;EAClB,aAAa,EAAE,IAAI;EACnB,gBAAgB,EAAE,wBAAwB;CAK3C;;AAHC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EAPrC,AAAA,MAAM,CAAC;IAQH,SAAS,EAAE,KAAK;GAEnB;;;AAED,AAAA,WAAW,CAAC;EACV,OAAO,EAAE,KAAK;EACd,KAAK,EAAE,IAAI;CACZ;;AAGC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,MAAM;EADtC,AAAA,SAAS,CAAC;IAEN,UAAU,EAAE,GAAG;GAElB;;;AAGC,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EADrC,AAAA,YAAY,CAAC;IAET,UAAU,EAAE,IAAI;IAChB,UAAU,EAAE,IAAI;GAEnB;;;AAED,AAAA,MAAM,CAAC;EACL,eAAe,EAAE,SAAS;CAC3B",
    "sources": [
        "UserProfile.module.scss",
        "../../shared/scss/variables.scss"
    ],
    "names": [],
    "file": "UserProfile.module.css"
}