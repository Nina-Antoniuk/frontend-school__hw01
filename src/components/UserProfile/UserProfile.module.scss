@import '../../shared/scss/variables.scss';

.userProfile {
  padding: 20px;
  box-shadow: var(--shadow) 0px 5px 15px;
  background-color: var(--light-color);

  @media screen and (max-width: $breakpoint-max-mobile) {
    max-width: 300px;
    margin-left: auto;
    margin-right: auto;
    margin-bottom: 30px;
  }

  @media screen and (min-width: $breakpoint-min-tablet) {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    max-width: 500px;
    margin-left: 30%;
    margin-bottom: 50px;
    padding: 80px 50px 30px;
  }

  @media screen and (min-width: $breakpoint-min-desktop) {
    max-width: 800px;
  }
}

.thumb {
  max-width: 250px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 20px;
  background-color: var(--placeholder-color);

  @media screen and (min-width: $breakpoint-min-tablet) {
    max-width: 300px;
  }
}

.userAvatar {
  display: block;
  width: 100%;
}

.userInfo {
  @media screen and (min-width: $breakpoint-min-desktop) {
    flex-basis: 50%;
  }
}

.description {
  @media screen and (min-width: $breakpoint-min-tablet) {
    flex-basis: 100%;
    margin-top: 30px;
  }
}

.field {
  text-decoration: underline;
}
