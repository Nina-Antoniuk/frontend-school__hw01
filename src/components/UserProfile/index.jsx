import React from 'react';
import { PropTypes } from 'prop-types';
import styles from './UserProfile.module.scss';

const UserProfile = function UserProfile({
  user: { avatarMedium, nickname, signature },
}) {
  return (
    <div className={styles.userProfile}>
      <div className={styles.thumb}>
        <img
          data-testid="userAvatar"
          className={styles.userAvatar}
          src={avatarMedium || 'https://via.placeholder.com/300x300'}
          alt={nickname || 'placeholder for stranger'}
          width="300"
          height="300"
        />
      </div>
      <div className={styles.userInfo}>
        <p>
          <span className={styles.field}>User name</span>:
          <span>{nickname || 'stranger'}</span>
        </p>
        <p>
          <span className={styles.field}>Signature</span>:{signature || ''}
        </p>
      </div>
    </div>
  );
};

UserProfile.defaultProps = {
  user: {
    nickname: '',
    avatarMedium: '',
    signature: '',
  },
};

UserProfile.propTypes = {
  user: PropTypes.shape({
    avatarMedium: PropTypes.string,
    nickname: PropTypes.string,
    signature: PropTypes.string,
  }),
};

export default UserProfile;
