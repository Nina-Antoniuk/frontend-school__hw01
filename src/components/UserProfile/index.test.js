import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import '@testing-library/jest-dom';
import UserProfile from './index';

let container = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const fakeProps = {
  user: {
    nickname: 'Test nickname',
    avatarMedium: 'test src',
    signature: 'test signature',
  },
};

test('TukTukApp', () => {
  act(() => {
    render(<UserProfile {...fakeProps} />, container);
  });
  expect(container).toMatchSnapshot();
});

// describe('component contain props', () => {
//   act(() => {
//     render(<UserProfile {...fakeProps} />, container);
//   });
//   test('nickname', () => {
//     expect(container.textContent).toContain(fakeProps.user.nickname);
//   });
//   test('signature', () => {
//     expect(container.querySelector('.userInfo').textContent).toContain(
//       fakeProps.user.signature,
//     );
//   });
// });
