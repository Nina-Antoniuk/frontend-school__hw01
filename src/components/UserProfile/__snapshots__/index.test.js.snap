// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TukTukApp 1`] = `
<div>
  <div
    class="userProfile"
  >
    <div
      class="thumb"
    >
      <img
        alt="Test nickname"
        class="userAvatar"
        height="300"
        src="test src"
        width="300"
      />
    </div>
    <div
      class="userInfo"
    >
      <p>
        <span
          class="field"
        >
          User name
        </span>
        :
        Test nickname
      </p>
      <p>
        <span
          class="field"
        >
          Signature
        </span>
        :
        test signature
      </p>
    </div>
  </div>
</div>
`;
